framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -1159,23 +1162,36 @@ protected void processConditionalPostconditions(
      * Add information from the postconditions and conditional postconditions of a method to the
      * stores after an invocation.
      *
-     * @param invocationNode a method call
-     * @param invocationTree the tree for the method call
+     * @param n a method call node or an object creation node
+     * @param tree the tree for the method call or for the object creation
      * @param thenStore the "then" store; is side-effected by this method
      * @param elseStore the "else" store; is side-effected by this method
      * @param postconditions the postconditions
      */
     private void processPostconditionsAndConditionalPostconditions(
-            MethodInvocationNode invocationNode,
-            Tree invocationTree,
+            Node n,
+            ExpressionTree tree,
             S thenStore,
             S elseStore,
             Set<? extends Contract> postconditions) {
 
-        StringToJavaExpression stringToJavaExpr =
-                stringExpr ->
-                        StringToJavaExpression.atMethodInvocation(
-                                stringExpr, invocationNode, analysis.checker);
+        StringToJavaExpression stringToJavaExpr = null;
+        if (n instanceof MethodInvocationNode) {
+            stringToJavaExpr =
+                    stringExpr ->
+                            StringToJavaExpression.atMethodInvocation(
+                                    stringExpr, (MethodInvocationNode) n, analysis.checker);
+        } else if (n instanceof ObjectCreationNode) {
+            stringToJavaExpr =
+                    stringExpr ->
+                            StringToJavaExpression.atConstructorInvocation(
+                                    stringExpr, (NewClassTree) tree, analysis.checker);
+        } else {
+            throw new BugInCF(
+                    "Error passing other type of nodes into"
```

Comment:
Try to make the error message more useful: what node and what kind of node were passed?

Reply:
Thanks!